src/components/home/elements/SendToken.vue

```diff
@@ -0,0 +1,960 @@
+        }
+    },
+    data: function() {
+        return initData
+    },
+    created() {
+        this.coldRecipientAddressList = new LRUCache(10)
+        this.hotRecipientAddressList = new LRUCache(10)
+        let item = window.localStorage.getItem('Cold ' + this.defaultColdAddress + ' sendTokenRecipientAddressList ')
+        if (item) {
+            this.coldRecipientAddressList.load(JSON.parse(item))
+        }
+        item = window.localStorage.getItem('Hot ' + this.defaultAddress + ' sendTokenRecipientAddressList ')
+        if (item) {
+            this.hotRecipientAddressList.load(JSON.parse(item))
+        }
+    },
+    computed: {
+        contractId() {
+            return transaction.tokenIDToContractID(this.tokenId)
+        },
+        defaultAddress() {
+            return Vue.ls.get('address')
+        },
+        defaultColdAddress() {
+            if (this.noColdAddress) return ''
+            return Object.keys(this.coldAddresses)[0]
+        },
+        userInfo() {
+            return JSON.parse(window.localStorage.getItem(this.defaultAddress))
+        },
+        secretInfo() {
+            return JSON.parse(
+                seedLib.decryptSeedPhrase(this.userInfo.info, Vue.ls.get('pwd')))
+        },
+        seedPhrase() {
+            return seedLib.decryptSeedPhrase(this.secretInfo.encrSeed, Vue.ls.get('pwd'))
+        },
+        wordList() {
+            return this.seedPhrase.split(' ')
+        },
+        isSubmitDisabled() {
+            return !(this.recipient && BigNumber(this.amount).isGreaterThan(0) && this.isValidRecipient(this.recipient) && (this.isValidAttachment || !this.attachment) && this.isAmountValid('hot') && this.address !== '')
+        },
+        isColdSubmitDisabled() {
+            return !(this.coldAddress && this.coldRecipient && this.coldAmount > 0 && this.isValidRecipient(this.coldRecipient) && (this.isValidColdAttachment || !this.coldAttachment) && this.isAmountValid('cold') && this.coldAddress !== '')
+        },
+        noColdAddress() {
+            return Object.keys(this.coldAddresses).length === 0 && this.coldAddresses.constructor === Object
+        },
+        dataObject() {
+            return {
+                protocol: PROTOCOL,
+                api: this.coldApi(),
+                opc: OPC_FUNCTION,
+                address: this.coldAddress,
+                senderPublicKey: this.coldAddresses[this.coldAddress].publicKey,
+                fee: this.coldFee * VSYS_PRECISION,
+                feeScale: FEE_SCALE,
+                timestamp: Date.now(),
+                attachment: transaction.prepareSendAttachment(this.coldAttachment),
+                contractId: this.contractId,
+                functionId: this.functionIndex,
+                function: transaction.prepareSend(this.coldRecipient, BigNumber(this.coldAmount).multipliedBy(this.tokenUnity)),
+                functionExplain: 'send ' + this.coldAmount + ' token to ' + this.coldRecipient
```

Comment:
how about the case amount = 1.00001 and unity = 1000 ?

Reply:
this to all usage above

Reply:
Tested this case. It cannot be passed.
<img width="455" alt="Screenshot 2019-05-23 at 1 39 27 PM" src="https://user-images.githubusercontent.com/41148850/58228133-3588c900-7d60-11e9-9e15-095d7332a54d.png">